presto-cache/src/main/java/com/facebook/presto/cache/alluxio/AlluxioCachingClientFileSystem.java

```diff
@@ -0,0 +1,255 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.cache.alluxio;
+
+import alluxio.AlluxioURI;
+import alluxio.client.file.FileInStream;
+import alluxio.client.file.FileOutStream;
+import alluxio.client.file.URIStatus;
+import alluxio.conf.AlluxioConfiguration;
+import alluxio.grpc.CreateDirectoryPOptions;
+import alluxio.grpc.CreateFilePOptions;
+import alluxio.grpc.DeletePOptions;
+import alluxio.grpc.ExistsPOptions;
+import alluxio.grpc.FreePOptions;
+import alluxio.grpc.GetStatusPOptions;
+import alluxio.grpc.ListStatusPOptions;
+import alluxio.grpc.MountPOptions;
+import alluxio.grpc.OpenFilePOptions;
+import alluxio.grpc.RenamePOptions;
+import alluxio.grpc.ScheduleAsyncPersistencePOptions;
+import alluxio.grpc.SetAclAction;
+import alluxio.grpc.SetAclPOptions;
+import alluxio.grpc.SetAttributePOptions;
+import alluxio.grpc.UnmountPOptions;
+import alluxio.security.authorization.AclEntry;
+import alluxio.security.authorization.Mode;
+import alluxio.wire.BlockLocationInfo;
+import alluxio.wire.FileInfo;
+import alluxio.wire.MountPointInfo;
+import alluxio.wire.SyncPointInfo;
+import com.facebook.presto.hive.filesystem.ExtendedFileSystem;
+import org.apache.hadoop.fs.BlockLocation;
+import org.apache.hadoop.fs.FSDataInputStream;
+import org.apache.hadoop.fs.FileStatus;
+import org.apache.hadoop.fs.Path;
+import org.apache.hadoop.fs.permission.FsPermission;
+
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+import static com.google.common.hash.Hashing.md5;
+import static java.nio.charset.StandardCharsets.UTF_8;
+import static java.util.Objects.requireNonNull;
+
+public class AlluxioCachingClientFileSystem
+        implements alluxio.client.file.FileSystem
+{
+    private final AlluxioConfiguration alluxioConfiguration;
+    protected final ExtendedFileSystem fileSystem;
+
+    public AlluxioCachingClientFileSystem(
+            ExtendedFileSystem fileSystem,
+            AlluxioConfiguration alluxioConfiguration)
+    {
+        this.fileSystem = requireNonNull(fileSystem, "fileSystme is null");
+        this.alluxioConfiguration = requireNonNull(alluxioConfiguration, "alluxioConfiguration is null");
+    }
+
+    @Override
+    public void createDirectory(AlluxioURI alluxioURI, CreateDirectoryPOptions options)
+            throws IOException
+    {
+        FsPermission permission = new FsPermission(Mode.fromProto(options.getMode()).toShort());
+        fileSystem.mkdirs(toPath(alluxioURI), permission);
+    }
+
+    @Override
+    public FileInStream openFile(AlluxioURI alluxioURI, OpenFilePOptions options)
+            throws IOException
+    {
+        FSDataInputStream inputStream = fileSystem.open(toPath(alluxioURI));
```

Comment:
Will this be a problem given we always force to use `FSDataInputStream openFile(Path path, HiveFileContext hiveFileContext)` to open files?

Reply:
Not sure if I understand your question?

Reply:
This patch is not using this interface: `openFile(Path path, HiveFileContext hiveFileContext)`. We need to tunnel through `hiveFileContext` from AlluxioCachingFS to the actual data tier. Otherwise ExtendedFileSystem work is not useful.

Reply:
@apc999 if you notice my earlier PR, they were using the AlluxioURIStatus to pass the information around.

Reply:
@jainxrohit I don't see it from https://github.com/prestodb/presto/pull/14196/files, can you point it to me?

Reply:
I have fixed it.